src/main/java/uk/nhs/hee/tis/trainee/forms/service/LtftService.java

```diff
@@ -89,4 +92,73 @@ public long getAdminLtftCount(Set<LifecycleState> states) {
 
     return ltftFormRepository.countByStatus_CurrentIn(states);
   }
+
+  /**
+   * Get the 'id' LTFT form if it belongs to the current user.
+   *
+   * @return The LTFT form, or optional empty if not found or does not belong to user.
+   */
+  public Optional<LtftFormDto> getLtftForm(UUID formId) {
+    String traineeId = traineeIdentity.getTraineeId();
+    log.info("Getting LTFT form {} for trainee [{}]", formId, traineeId);
+
+    Optional<LtftForm> form = ltftFormRepository.findByTraineeTisIdAndId(traineeId, formId);
+    if (form.isPresent()) {
+      log.info("Found form {} for trainee [{}]", formId, traineeId);
+      return Optional.of(mapper.toDto(form.get()));
+    } else {
+      log.info("Did not find form {} for trainee [{}]", formId, traineeId);
+      return Optional.empty();
+    }
+  }
+
+  /**
+   * Save the dto as a new LTFT form.
+   *
+   * @param dto The LTFT DTO to save.
+   * @return The saved form DTO.
+   */
+  public Optional<LtftFormDto> saveLtftForm(LtftFormDto dto) {
+    String traineeId = traineeIdentity.getTraineeId();
+    log.info("Saving LTFT form for trainee [{}]: {}", traineeId, dto);
+    LtftForm form = mapper.toEntity(dto);
+    if (!form.getTraineeTisId().equals(traineeId)) {
+      log.warn("Could not save form since it does belong to the logged-in trainee {}: {}",
+          traineeId, dto);
+      return Optional.empty();
+    }
+    LtftForm savedForm = ltftFormRepository.save(form);
+    return Optional.of(mapper.toDto(savedForm));
+  }
+
+  /**
+   * Update the existing LTFT form.
+   *
+   * @param formId The id of the LTFT form to update.
+   * @param dto    The updated LTFT DTO to save.
+   * @return The updated form DTO.
+   */
+  public Optional<LtftFormDto> updateLtftForm(UUID formId, LtftFormDto dto) {
+    String traineeId = traineeIdentity.getTraineeId();
+    log.info("Updating LTFT form {} for trainee [{}]: {}", formId, traineeId, dto);
+    LtftForm form = mapper.toEntity(dto);
+    if (form.getId() == null || !form.getId().equals(formId)) {
+      log.warn("Could not update form since its id {} does not equal provided form id {}",
+          form.getId(), formId);
+      return Optional.empty();
+    }
+    if (!form.getTraineeTisId().equals(traineeId)) {
+      log.warn("Could not update form since it does belong to the logged-in trainee {}: {}",
+          traineeId, dto);
+      return Optional.empty();
+    }
+    Optional<LtftForm> existingForm = ltftFormRepository.findByTraineeTisIdAndId(traineeId, formId);
+    if (existingForm.isEmpty()) {
+      log.warn("Could not update form {} since no existing form with this id for trainee {}",
+          formId, traineeId);
+      return Optional.empty();
+    }
+    LtftForm savedForm = ltftFormRepository.save(form);
+    return Optional.of(mapper.toDto(savedForm));
```

Comment:
As above.

Reply:
savedForm is not an Optional<>, so it doesn't have the map unfortunately. It feels we're not gaining much by making it 
```
Optional<LtftForm> savedForm = Optional.of(ltftFormRepository.save(form));
return savedForm.map(mapper::toDto)
```

Reply:
I misread, sorry!